src/ChurchCRM/Emails/BaseEmail.php

```diff
@@ -111,9 +109,9 @@ protected function getCommonTokens(): array
         return $commonTokens;
     }
 
-    abstract public function getTokens();
+    abstract public function getTokens(): array;
```

Comment:
can you typehint the shape of the array in the docblock?

Reply:
How do you do that

Reply:
```suggestion
    /**
     * @return array<string, string>
     */
    abstract public function getTokens(): array;
```

https://phpstan.org/writing-php-code/phpdocs-basics#combining-phpdoc-types-with-native-typehints - typehints like this helps when developing so users know what to expect from a given function.